src/sentry/receivers/features.py

```diff
@@ -0,0 +1,189 @@
+from __future__ import absolute_import
+
+from sentry.adoption import manager
+from sentry.interfaces.base import get_interface
+from sentry.models import FeatureAdoption
+from sentry.plugins import IssueTrackingPlugin, IssueTrackingPlugin2
+from sentry.plugins.bases.notify import NotificationPlugin
+from sentry.receivers.rules import DEFAULT_RULE_LABEL, DEFAULT_RULE_DATA
+from sentry.signals import (
+    alert_rule_created,
+    event_processed,
+    first_event_received,
+    project_created,
+    member_joined,
+    plugin_enabled,
+    user_feedback_received,
+    issue_assigned,
+    issue_resolved_in_release,
+    advanced_search,
+    save_search_created,
+    inbound_filter_toggled,
+    sso_enabled,
+    data_scrubber_enabled,
+)
+from sentry.utils.javascript import has_sourcemap
+
+DEFAULT_TAGS = frozenset(['level', 'logger', 'transaction', 'url', 'browser', 'sentry:user',
+    'os', 'server_name', 'device', 'os.name', 'browser.name', 'sentry:release', 'environment',
+    'device.family', 'site', 'version', 'interface_type', 'rake_task', 'runtime', 'runtime.name',
+    'type', 'php_version', 'app', 'app.device', 'locale', 'os_version', 'device_model',
+    'deviceModel', 'sentry_version'])
+
+
+# First Event
+@first_event_received.connect(weak=False)
+def record_first_event(project, group, **kwargs):
+    FeatureAdoption.objects.record(
+        organization_id=project.organization_id,
+        feature_slug="first_event",
+        complete=True)
+
+
+@event_processed.connect(weak=False)
+def record_event_processed(project, group, event, **kwargs):
+    feature_slugs = []
+
+    # Platform
+    if group.platform in manager.location_slugs('language'):
+        feature_slugs.append(group.platform)
+
+    elif event.data.get(get_interface('csp')):
+        feature_slugs.append('csp')
+
+    # Release Tracking
+    if event.get_tag('sentry:release'):
+        feature_slugs.append('release_tracking')
+
+    # Environment Tracking
+    if event.get_tag('environment'):
+        feature_slugs.append('environment_tracking')
+
+    # User Tracking
+    user_context = event.data.get('sentry.interfaces.User')
+    # We'd like them to tag with id or email.
+    # Certain SDKs automatically tag with ip address.
+    # Check to make sure more the ip address is being sent.
+    # testing for this in test_no_user_tracking_for_ip_address_only
+    # list(d.keys()) pattern is to make this python3 safe
+    if user_context and list(user_context.keys()) != ['ip_address']:
+        feature_slugs.append('user_tracking')
+
+    # Custom Tags
+    if set(tag[0] for tag in event.tags) - DEFAULT_TAGS:
+        feature_slugs.append('custom_tags')
+
+    # Sourcemaps
+    if has_sourcemap(event):
+        feature_slugs.append('source_maps')
+
+    # Breadcrumbs
+    if event.data.get('sentry.interfaces.Breadcrumbs'):
+        feature_slugs.append('breadcrumbs')
+
+    FeatureAdoption.objects.bulk_record(project.organization_id, feature_slugs)
```

Comment:
Before this, we should short circuit if there are no feature_slugs. It seems nothing is handling the case where there are none gracefully.

Alternatively, can add the check inside `bulk_record`. Just a simple `if not feature_slugs: return`